app/controllers/uploads_controller.rb

```diff
@@ -69,7 +69,7 @@ def lookup_urls
   end
 
   def show
-    return render_404 if !RailsMultisite::ConnectionManagement.has_db?(params[:site])
+    return render_404 if !RailsMultisite::ConnectionManagement.has_db?(params[:site]) || request.xhr?
```

Comment:
I think it would be good to add an explanation comment here, because it's not obvious why `||request.xhr?` is included